lib/puppet/provider/postgresql_conf/ruby.rb

```diff
@@ -0,0 +1,160 @@
+# frozen_string_literal: true
+
+# This provider is used to manage postgresql.conf files
+# It uses ruby to parse the config file and
+# to add, remove or modify settings.
+#
+# The provider is able to parse postgresql.conf files with the following format:
+# key = value # comment
+
+Puppet::Type.type(:postgresql_conf).provide(:ruby) do
+  desc 'Set keys, values and comments in a postgresql config file.'
+  confine kernel: 'Linux'
+
+  # file parsing in a seperate function
```

Comment:
I had a chat with @ekohl about it, I think we should use YARD to document the methods (see the line method for examples)

Reply:
please check if thats ok